File: Whatsapp-OCR-CRM/backend/services/activity.py
"""Activity logging helper."""
from database import db
from models import ActivityLog


async def log_activity(user_id: str, action: str, entity_type: str, entity_id: str = None):
    if not user_id:
        return
    a = ActivityLog(userId=user_id, action=action, entityType=entity_type, entityId=entity_id)
    await db.activity_logs.insert_one(a.model_dump())

File: Whatsapp-OCR-CRM/backend/src/utils/activity.ts
import { prisma } from "../lib/prisma";
import { logger } from "./logger";

export async function logActivity(userId: string, action: string, entityType: string, entityId: string) {
  if (!userId) return;
  try {
    await prisma.activityLog.create({
      data: {
        userId,
        action,
        entityType,
        entityId,
      },
    });
  } catch (error) {
    logger.error("Failed to log activity: " + error);
  }
}
